addons/survey/static/tests/components/survey_question_trigger_widget_tests.js

```diff
@@ -0,0 +1,169 @@
+                    fields: {
+                        name: {type: "char", string: "name"},
+                    },
+                    records: [
+                        {
+                            id: 1,
+                            name: "Answer 1"
+                        },
+                    ]
+                }
+            },
+            views: {
+                "survey_question,false,form": `
+                    <form>
+                        <group>
+                            <field name="name"/>
+                            <field name="triggering_answer_id" invisible="1"/>
+                            <field name="triggering_question_id" invisible="1"/> 
+                            <widget name="survey_question_trigger" nolabel="1"/>
+                        </group>
+                    </form>
+                `,
+            },
+        };
+
+        setupViewRegistries();
+    });
+
+    QUnit.test("dynamic rendering of surveyQuestionTriggerError rows", async (assert) => {
+        await makeView({
+            type: "form",
+            resModel: "survey",
+            resId: 1,
+            serverData,
+            arch: `
+                <form>
+                    <!-- "question_page_one2many" uses "QuestionPageListRenderer" that implements "getSurveyQuestionTriggerError"-->
+                    <field name="question_and_page_ids" widget="question_page_one2many">  
+                        <tree>
+                            <field name="sequence" widget="handle"/>
+                            <field name="name" />
+                            <field name="triggering_answer_id" invisible="1"/>
+                            <field name="triggering_question_id" invisible="1"/> 
+                            <widget name="survey_question_trigger" nolabel="1"/>
+                        </tree>
+                    </field>
+                </form>
+            `,
+        });
+
+        assert.containsOnce(target, ".o_field_x2many .o_list_renderer table.o_section_list_view");
+        assert.containsN(target, ".o_data_row", 2);
+        let rows = target.querySelectorAll(".o_data_row");
+
+        assert.strictEqual(rows[0].textContent, "Question 1");
+        let q1TriggerDiv = rows[0].querySelector("td.o_data_cell div.o_widget_survey_question_trigger");
+        assert.containsNone(q1TriggerDiv, "button");
+
+        assert.strictEqual(rows[1].textContent, "Question 2");
+        let q2TriggerDiv = rows[1].querySelector("td.o_data_cell div.o_widget_survey_question_trigger");
+        assert.containsOnce(q2TriggerDiv, "button");
+        // Question 2 is correctly placed after Question 1
+        let triggerIcon = q2TriggerDiv.querySelector("button i");
+        assert.doesNotHaveClass(triggerIcon, "text-warning");
+        assert.hasAttrValue(triggerIcon, 'data-tooltip', 'Displayed if "Question 1: Answer 1"',
+                       'Trigger tooltip should be \'Displayed if "Question 1: Answer 1"\'.');
```

Comment:
When assertion text does not explain "why" and just repeats what the error will say, I usually don't bother putting it

(Same for others)

Reply:
I see what you mean, did not notice you have twice the same test.

Even though I think the error message mentions the line number, but you can keep it if you want :)